### PR TITLE
Formal parameters sync without reloading

### LINKED ISSUE #4544: Formal parameters suggestions are not reloaded until view is reopened
### Describe the bug

Variables are only loaded when you open the BKM, so you need to reopen the view after you add the formal parameters:

https://github.com/user-attachments/assets/2011b103-531a-44c9-be54-571d700ede5c

### Steps to reproduce

1. Create a BKM
2. Add formal parameters
3. Try to use the suggestions

### Expected behavior

Suggestions are always up to date.

### Environment

- OS: [e.g. MacOS 10.2, Windows 10]
- Camunda Modeler Version: [e.g. 2.0.0]
- Execution Platform: [e.g. Camunda 7, Camunda 8]
- Installed plug-ins: [...]

### Additional context

Initially reported in https://github.com/bpmn-io/dmn-js/issues/896

### PR DESCRIPTION
**Pull request by abdul99ahad**
Closes #4544 

### Proposed Changes

 
**Previously**: Formal parameters suggestions were not available in the literal expression until you reload the view
**Changes**: Formal parameters suggestions are available without reloading

Nov-19-2024 14-27-48

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] __Brief textual description__ of the changes present
* [x] __Visual demo__ attached
* [ ] __Steps to try out__ present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ ___Note:__ Yet to be released changes appear here._
 * `FEAT`: make variables tab accessible via the application footer ([#4516](https://github.com/camunda/camunda-modeler/issues/4516))
 * `FIX`: prevent long element name from shrinking icon ([#4505](https://github.com/camunda/camunda-modeler/issues/4505))
 * `FIX`: correct default extension used when saving Camunda 8 BPMN diagrams ([#4661](https://github.com/camunda/camunda-modeler/issues/4661))
+* `FIX`: formal parameter suggestions sync without requiring view refresh ([#4544](https://github.com/camunda/camunda-modeler/issues/4544))
```
Comment (philippfromme) [inline CHANGELOG.md:22]: This is a DMN change so it should be in the ## DMN section.

**Review by philippfromme — COMMENTED** [review]
Works. Let's fix the changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,11 @@ ___Note:__ Yet to be released changes appear here._
 * `FIX`: prevent long element name from shrinking icon ([#4505](https://github.com/camunda/camunda-modeler/issues/4505))
 * `FIX`: correct default extension used when saving Camunda 8 BPMN diagrams ([#4661](https://github.com/camunda/camunda-modeler/issues/4661))
 
+### DMN
+
+* `FIX`: formal parameter suggestions sync without requiring view refresh ([#4544](https://github.com/camunda/camunda-modeler/issues/4544))
+* `DEPS`: update to `camunda-dmn-js@3.0.0`
```
Comment (philippfromme) [inline CHANGELOG.md:26]: The dependency change should go in the ### General group.

**Comment by nikku** [comment]
Closing in favor of https://github.com/camunda/camunda-modeler/pull/4620.

**Pull request closed without merging**